src/map/map.hpp

```diff
@@ -794,6 +806,90 @@ struct iwall_data {
 	bool shootable;
 };
 
+struct s_map_zone_data {
+	uint16 id;
+	std::unordered_map<std::string, uint16> disabled_commands;
+	std::unordered_map<uint16, std::pair<uint16, uint16>> disabled_skills;
+	std::unordered_map<t_itemid, uint16> disabled_items;
+	std::unordered_map<sc_type, uint16> disabled_statuses;
+	std::unordered_map<int32, uint16> restricted_jobs;
+
+	bool isCommandDisabled(std::string name, uint16 group_lv) {
+		if (this->disabled_commands.empty())
+			return false;
+
+		auto cmd_lv = util::umap_find(this->disabled_commands, name);
```

Comment:
Based on the calls to this and the other functions below:
My suggestion would be to hand "sd" over and check if it is nullptr and do the according checks in here.